cdap-test/src/main/java/co/cask/cdap/test/TestManager.java

```diff
@@ -45,6 +50,37 @@ ApplicationManager deployApplication(Id.Namespace namespace,
                                        Class<? extends Application> applicationClz, File... bundleEmbeddedJars);
 
   /**
+   * Deploys an {@link ApplicationTemplate}. Only supported in unit tests.
+   *
+   * @param namespace The namespace to deploy to
+   * @param templateClz The template class
+   * @param templateId The id of the template. Must match the name set in
+   *                   {@link ApplicationTemplate#configure(ApplicationConfigurer, ApplicationContext)}
+   */
+  void deployTemplate(Id.Namespace namespace, Id.ApplicationTemplate templateId,
+                      Class<? extends ApplicationTemplate> templateClz) throws IOException;
+
+  /**
+   * Adds a plugin usable by the given template. Only supported in unit tests.
+   *
+   * @param templateId The id of the template to add the plugin for
+   * @param pluginClz The plugin class
+   * @param jarName The name to use for the plugin jar
+   * @throws IOException
+   */
+  void addTemplatePlugin(Id.ApplicationTemplate templateId, Class<?> pluginClz, String jarName) throws IOException;
+
+  /**
+   * Creates an adapter.
+   *
+   * @param adapterId The id of the adapter to create
+   * @param config The configuration for the adapter
+   * @return An {@link AdapterManager} to manage the deployed adapter.
+   * @throws Exception if there was an exception deploying the adapter.
+   */
+  AdapterManager deployAdapter(Id.Adapter adapterId, AdapterConfig config) throws Exception;
```

Comment:
hm yeah i think that is a better name since we're not deploying a jar or anything. will change.